libindi/drivers/telescope/celestrondriver.cpp

```diff
@@ -857,8 +862,16 @@ bool CelestronDriver::foc_limits(int * low, int * high)
     if (rlen < 8)
         return false;
 
-    *low = (response[0] << 24) + (response[1] << 16) + (response[2] << 8) + response[3];
-    *high = (response[4] << 24) + (response[5] << 16) + (response[6] << 8) + response[7];
+    *low = ((uint8_t)response[0] << 24) + ((uint8_t)response[1] << 16) + ((uint8_t)response[2] << 8) + (uint8_t)response[3];
```

Comment:
how does the cast here affect the results? it is shifting bits, does it matter what the type is? bits are bits. Am I missing something?

Reply:
This is where the problem is, some compilers convert the signed char to a signed integer with the same value, then do the shift.
For example 0x80 is converted to 0xFFFFFF80, then an 8 bit shift gives 0xFFFF8000.

What's wanted is that 0x80 is shifted to 0x8000, this is what the RasbPi compiler does, which is why I didn't see it.
penguin's log shows this, it's OK with the Celestron SCT because the buffer uses unsigned char aka uint8_t but Celestron GPS uses a char array and it goes wrong for him on that one only. He's connected twice, using both celestron SCT and Celestron GPS.

I'm going to look at changing the type cast as you suggested.